stats/stats.go

```diff
@@ -203,24 +214,54 @@ func NewHistogram(Offset float64, Divider float64) *Histogram {
 	return h
 }
 
-// Tradeoff memory for speed (though that also kills the cache so...)
-// this creates an array of 100k (max value) entries
-// TODO: consider using an interval search for the last N big buckets
+// Val2Bucket values are kept in two different structure
+// constantTimeVal2Bucket allows you reach between 0 and 1000 in constant time
+// linearTimeVal2Bucket allows you to reach between 1000 and 100000 in linear time
+// linearTimeVal2Bucket approach saves memory usage but loses speed
```

Comment:
btw we can add a new TODO to do a binary search instead of linear (though that's maybe overkill depending on benchmark)